### DISCUSSION TITLE
Migrating from ZStack to Ember

### DISCUSSION THREAD

**Discussion by DJBenson** [Koenkk/zigbee2mqtt#23024] [General]
So I know it's not possible to do anything automatically and re-pairing is necessary but I just wanted to find out if anybody had moved from a ZStack (TI) to Ember (SiLabs) based solution and what steps are required above what the docs say.

I have about 50 devices on an Electrolama zzh stick which is connected to my virtualised (Proxmox) HAOS but it keeps disconnecting for some reason so I've got a SLZB-06M ready to go but just not plucked up the courage to do it yet.

The instructions seem fairly straight forward in that you stop Z2M, update the serial config, remove a couple of files, re-start Z2M and start the re-pairing process.

My concern is around entity ID's in HA - do these remain the same? I'm guessing as Z2M identifies devices by their IEEE address once the device appears on the "new" network it will still identify itself to HA as the same device?

If I begin the process and re-pair a couple of non-critical devices and it's not working out, is rolling back as simple as restoring the deleted files and re-pairing the devices which were moved across?

**Comment by DJBenson** [comment]
As a secondary question, I've seen some mention of migrating ZStack to EZSP using some python tools whose name eludes me right now, but as we've moved from EZSP to Ember I wondered if there was a migration path there?

**Comment by wojciechczyz** [comment]
It seems that repairing is NO LONGER necessary:
https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another

How ever, it is still not GUARANTEED. Did you tried it recently?
@Koenkk can we have it guaranteed? I am moving from ZStack to Ember as many people are due to superior memory and CPU in Ember sticks. How to improve chances of migration being successful?

**Comment by sigma415** [comment]
Has anyone successfully migrated from ZStack to Ember using the method described in the FAQ ? 
With a lot of repairing, or less ?

**Comment by richlawson** [comment]
Hi @sigma415 , in case it helps, I followed https://community.home-assistant.io/t/zigbee2mqtt-zbt-2/955316/2 back in December to migrate from a SONOFF Zigbee 3.0 USB (ZStack) Dongle Plus to a Home Assistant Connect ZBT-2 (Ember). 

I didn't have to re-pair any of my 64 Zigbee devices.
